dask/dataframe/optimize.py

```diff
@@ -20,8 +13,10 @@ def optimize(dsk, keys, **kwargs):
     try:
         from castra import Castra
         dsk3 = fuse_getitem(dsk2, Castra.load_partition, 3)
+        dsk4 = fuse_selections(dsk3, getattr, Castra.load_partition,
+                               lambda a, b: (Castra.load_index, b[1], b[2]))
```

Comment:
I disagree - this only matches graphs of the form:

```
{'key1': (getattr, 'key2', 'index'),
'key2': (Castra.load_partition, castra, name, columns)}
```

which should only occur for the cases we want.